src/transformers/generation/configuration_utils.py

```diff
@@ -574,17 +574,163 @@ def validate(self, is_init=False):
         """
 
         # Validation of individual attributes
+        pos_int_args = [
+            "top_k",
+            "bos_token_id",
+            "forced_bos_token_id",
+            "max_new_tokens",
+            "min_new_tokens",
+            "max_length",
+            "min_length",
+            "no_repeat_ngram_size",
+            "encoder_no_repeat_ngram_size",
+        ]
+
+        probability_args = ["top_p", "typical_p", "min_p", "epsilon_cutoff", "eta_cutoff"]
+
+        float_pos_args = [
+            "temperature",
```

Comment:
temperature should be in probability args

Reply:
It seems that `temperature` shouldn’t be bound to a [0, 1] range because it scales down the logits (either dividing by a value greater than 1 or less than 1 to control randomness), but it doesn’t fit in `float_pos_args` either as `temperature`  can not be zero; it should be strictly float positive so it might need a separate test :)

Reply:
right, let's leave it as float positive argument then as the logit processor is added in all cases when `temp != 1`. In case users sets `0`, another error is raised when init the logit processor class